include/eve/arch/arm/as_register.hpp

```diff
@@ -17,7 +17,99 @@ namespace eve
+        constexpr bool signed_v = std::is_signed_v<Type>;
+
+        if constexpr( std::is_same_v<Type,float> )
+        {
+          using type = svfloat32_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( std::is_same_v<Type,double> )
+        {
+          using type = svfloat64_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( signed_v && sizeof(Type) == 1 )
+        {
+          using type = svint8_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( signed_v && sizeof(Type) == 2 )
+        {
+          using type = svint16_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( signed_v && sizeof(Type) == 4 )
+        {
+          using type = svint32_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( signed_v && sizeof(Type) == 8 )
+        {
+          using type = svint64_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( !signed_v && sizeof(Type) == 1 )
+        {
+          using type = svuint8_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( !signed_v && sizeof(Type) == 2 )
+        {
+          using type = svuint16_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( !signed_v && sizeof(Type) == 4 )
+        {
+          using type = svuint32_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+        else if constexpr( !signed_v && sizeof(Type) == 8 )
+        {
+          using type = svuint64_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+          return type{};
+        }
+      }
+    }
+
+    using type = decltype(find());
+    static_assert( !std::is_void_v<type>, "[eve arm sve] - Type is not usable in a SIMD register");
+  };
+
+  // ---------------------------------------------------------------------------------------------
+  // logical cases
+  template<typename Type, typename Size, sve_abi ABI>
+  struct  as_logical_register<Type, Size, ABI>
+  {
+    static constexpr auto find()
+    {
+      constexpr auto width = sizeof(Type)*Size::value*8;
+
+      if constexpr(width <= __ARM_FEATURE_SVE_BITS)
+      {
+        using type = svbool_t __attribute__((arm_sve_vector_bits(__ARM_FEATURE_SVE_BITS)));
+        return type{};
+      }
+    }
+
+    using type = decltype(find());
+    static_assert( !std::is_void_v<type>, "[eve arm sve] - Type is not usable in a SIMD register");
+  };
+}
+#endif
+
+#if defined(EVE_HW_ARM) || defined(EVE_HW_ARM_SVE)
```

Comment:
Wait, this is confusing.

I thought we decided either arm or arm sve.

Why is this an if?